source/extensions/filters/http/jwt_authn/matcher.cc

```diff
@@ -188,6 +189,43 @@ class PathSeparatedPrefixMatcherImpl : public BaseMatcherImpl {
 };
 } // namespace
 
+class PathMatchPolicyMatcherImpl : public BaseMatcherImpl {
+public:
+  PathMatchPolicyMatcherImpl(const RequirementRule& rule,
+                             Server::Configuration::CommonFactoryContext& context)
+      : BaseMatcherImpl(rule, context), uri_template_matcher_(createUriTemplateMatcher(rule)) {}
+
+  bool matches(const Http::RequestHeaderMap& headers) const override {
+    if (BaseMatcherImpl::matchRoute(headers) &&
+        uri_template_matcher_->match(headers.getPathValue())) {
+      ENVOY_LOG(debug, "Path match policy requirement '{}' matched.",
+                uri_template_matcher_->uriTemplate());
+      return true;
+    }
+
+    return false;
+  }
+
+private:
+  const Router::PathMatcherSharedPtr uri_template_matcher_;
+
+  Router::PathMatcherSharedPtr createUriTemplateMatcher(const RequirementRule& rule) {
```

Comment:
```suggestion
  static Router::PathMatcherSharedPtr createUriTemplateMatcher(const RequirementRule& rule) {
```

Reply:
commited your suggestion 👍

Reply:
I think the test that does not pass (`test/integration/quic_http_integration_test`) is unrelated to my changes.